### PR TITLE
[IMP] web: add more accurate test on onchange scenario

### PR DESCRIPTION
**Pull request by aab-odoo**
In a form view, have an x2many list with two fields A and B (e.g.
A is a many2one, and B is a char). There is an onchange on the
x2many that is triggered when the user sets A, and that sets B.

In this scenario, let's assume that the user first writes something
in B, but directly deletes it. Then, he sets A. In this situation,
B isn't updated with the value returned by the onchange.

Commit [1] is an attempt to fix this, but it actually broke what
has been done in [2]. However, tests of [2] still passed, because
they didn't accurately reproduce what happens in practice (outside
the test environment).

An input field widget listens to 'input' and 'change' events. It
commits its value when both occur, but in the 'input' case, the
action is debounced with a large value (so that it actually only
commits the value on 'change', i.e. when the field is focusset
out). In the tests, the debounce value is 0, so fields actually
commit their value directly when 'input' events occur. This is
something we may want to change in the future, to remove this
discrepancy.

This commit adds a test with a large debounce value, so that we
accurately test what happens in practice, and we make sure that
this won't be broken in an attempt of fixing something else.

[1] cdca11db2ec207363df1d4fb3c87014d9c931d25
[2] a71f22370217d2da6ad13bd29c2fa412a40ebe4c

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by aab-odoo** [comment]
robodoo rebase-ff

File: addons/web/static/tests/fields/basic_fields_tests.js [hunk 1]
```diff
@@ -1680,6 +1738,54 @@ QUnit.module('basic_fields', {
...
+                    </field>
+                </form>`,
+            async mockRPC(route, args) {
+                const result = this._super(...arguments);
+                if (args.method === "onchange") {
+                    await Promise.resolve(def);
+                }
+                return result;
+            },
+            fieldDebounce: 1000, // needed to accurately mock whan really happens
```
Comment (kebeclibre) [inline addons/web/static/tests/fields/basic_fields_tests.js:1771]: "what"

File: addons/web/static/tests/fields/basic_fields_tests.js [hunk 2]
```diff
@@ -1569,6 +1569,64 @@ QUnit.module('basic_fields', {
...
+            'should contain the default value');
+
+        def = testUtils.makeTestPromise();
+
+        await testUtils.fields.many2one.clickOpenDropdown('product_id');
+        await testUtils.fields.many2one.clickHighlightedItem('product_id');
+
+        // set foo before onchange
+        await testUtils.fields.editInput(form.$('input[name="foo"]'), "tralala");
+        assert.strictEqual(form.$('input[name="foo"]').val(), 'tralala',
```
Comment (kebeclibre) [inline addons/web/static/tests/fields/basic_fields_tests.js:1617]: shouldn't we assert that we triggered some kind of event or business code ?
this line potentially just re-asserts the change that editInput did (setting the value of the input node)

**Comment by kebeclibre** [comment]
robodoo r+

**Pull request closed without merging**